src/components/CertificateVerifyBlock/DetailedCertificateVerifyBlock/TokenVerifyBlock.js

```diff
@@ -0,0 +1,31 @@
+import React, { useState, useEffect } from "react";
+import { getTokenOwner, initializeToken } from "../../../services/token";
+import css from "./detailedCertificateBlock.scss";
+
+const TokenVerifyBlock = ({ document }) => {
+  const [tokenOwner, setTokenOwner] = useState("");
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    async function fetchTokenOwner() {
+      try {
+        await initializeToken(document);
+        const owner = await getTokenOwner();
```

Comment:
alternatively move this call up to the main page level which might make more sense - because we're passing in the document all the way in here, which to me seems like a contradiction to the existing design which only receives statuses as props

Reply:
moved up to parent. need to add loader/spinner because its still taking time. created a new issue for that. As its the UI improvement.